pkg/controller/admin/mobile_apps_test.go

```diff
@@ -103,73 +71,15 @@ func TestAdminMobileApps(t *testing.T) {
 	t.Run("lists_all", func(t *testing.T) {
 		t.Parallel()
 
-		mux := mux.NewRouter()
-		mux.Use(middlewares...)
-		mux.Handle("/", c.HandleMobileAppsShow()).Methods(http.MethodGet)
-
 		ctx := ctx
 		ctx = controller.WithSession(ctx, &sessions.Session{})
 		ctx = controller.WithUser(ctx, &database.User{})
 
-		r := httptest.NewRequest(http.MethodGet, "/", nil)
-		r = r.Clone(ctx)
-		r.Header.Set("Content-Type", "text/html")
-
-		w := httptest.NewRecorder()
-
-		mux.ServeHTTP(w, r)
-		w.Flush()
+		w, r := envstest.BuildFormRequest(ctx, t, http.MethodGet, "/", nil)
+		handler.ServeHTTP(w, r)
 
 		if got, want := w.Code, http.StatusOK; got != want {
 			t.Errorf("Expected %d to be %d", got, want)
 		}
 	})
-
-	t.Run("renders", func(t *testing.T) {
-		t.Parallel()
-
-		realm, _, session, err := harness.ProvisionAndLogin()
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		cookie, err := harness.SessionCookie(session)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		app := &database.MobileApp{
-			Name:    "test mobile app",
-			RealmID: realm.ID,
-			URL:     "https://example2.com",
-			OS:      database.OSTypeAndroid,
-			SHA:     "AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA:AA",
-			AppID:   "app2",
-		}
-		if err := harness.Database.SaveMobileApp(app, database.SystemTest); err != nil {
-			t.Fatal(err)
-		}
-
-		browserCtx := browser.New(t)
-		taskCtx, done := context.WithTimeout(browserCtx, project.TestTimeout())
-		defer done()
-
-		if err := chromedp.Run(taskCtx,
-			browser.SetCookie(cookie),
-
-			chromedp.Navigate(`http://`+harness.Server.Addr()+`/admin/mobile-apps`),
-			chromedp.WaitVisible(`body#admin-mobileapps-index`, chromedp.ByQuery),
-
-			chromedp.SetValue(`input#search`, "test mobile app", chromedp.ByQuery),
-			chromedp.Submit(`form#search-form`, chromedp.ByQuery),
-			chromedp.WaitVisible(`table#results-table tr`, chromedp.ByQuery),
-
-			chromedp.SetValue(`input#search`, "notexists", chromedp.ByQuery),
-			chromedp.Submit(`form#search-form`, chromedp.ByQuery),
```

Comment:
Wasn't this exercising more before - it actually runs a search.

Reply:
Fixed